### PR TITLE
Transfer Hook Docs

### LINKED ISSUE #4942: Update docs for Transfer Hook & TLV Account Resolution
### Problem
Now that PDA account resolution has been added to `spl-tlv-account-resolution`, we need some updated documentation!

### Solution
- Update the documentation for Transfer Hook
- Update the documentation for TLV Account Resolution
- Release new versions

### PR DESCRIPTION
**Pull request by buffalojoec**
This PR updates the documentation for the Transfer Hook Interface and the TLV Account Resolution library

Closes #4942

### CODE REVIEW

File: docs/src/token-2022/extensions.mdx [hunk 1]
```diff
@@ -1358,8 +1362,66 @@ at that instruction descriminator. Your program may implement multiple interface
 so any other instruction in your program can create the account at the program-derived
 address!
 
-More information in the
-[spl-transfer-hook-interface README](https://github.com/solana-labs/solana-program-library/tree/master/token/transfer-hook-interface/README.md).
+When your program stores configurations for extra required accounts in the
+well-defined program-derived address, it's possible to send an instruction -
+such as `Execute` (transfer) - to your program with only accounts required
+for the interface instruction, and all extra required accounts are
+automatically resolved!
```
Comment (joncinque) [inline docs/src/token-2022/extensions.mdx:1369]: Let's make it clearer how the resolution is happening
```suggestion
automatically resolved on-chain through helper functions!
```

File: docs/src/token-2022/extensions.mdx [hunk 2]
```diff
@@ -1358,8 +1362,66 @@ at that instruction descriminator. Your program may implement multiple interface
...
+**dynamic program-derived address** associated with their wallet via the
+following seeds: `"can_transfer" + <wallet_address>`.
+
+Using the transfer-hook interface, you can store these configurations in the
+well-defined program-derived address for your mint and program id.
+
+When a user attempts to transfer your token, they might provide to Token-2022:
+
+```rust
+[source, mint, destination, owner/delegate]
```
Comment (joncinque) [inline docs/src/token-2022/extensions.mdx:1413]: This isn't something to change in this PR, but this example outlines an issue with the current approach, and makes it clearer that we definitely need the ability to read into account data. Here, you give `owner/delegate`, but if it's the delegate, than the address won't be the same as with `owner`

File: docs/src/token-2022/extensions.mdx [hunk 3]
```diff
@@ -1347,6 +1347,10 @@ A usable example program exists at
 Token-2022 uses this example program in tests to ensure that it properly uses
 the transfer hook interface.
 
+The example program and the interface are powered by the
+[spl-tlv-account-resolution](https://github.com/solana-labs/solana-program-library/tree/master/libraries/tlv-account-resolution/README.md)
+library, which is explained a bit more below.
+
 #### How to use
```
Comment (joncinque) [inline docs/src/token-2022/extensions.mdx:1354]: There's a lot of copy-pasta between all these sections -- how about just linking these to the sections in the README with one sentence?

File: token/transfer-hook-interface/README.md [hunk 4]
```diff
@@ -79,28 +70,92 @@ harder.
...
+  - Hard-coded values, such as string literals or integers
+  - A slice of the instruction data provided to the transfer-hook program
+  - The address of another account in the total list of accounts
+
+When you store configurations for a dynamic Program-Derived Address within the
+additional required accounts, the PDA itself is evaluated (or resolved) at the
+time of instruction invocation using the instruction itself. This
+occurs in the offchain and onchain helpers mentioned below, which leverage
+the SPL TLV Account Resolution library to perform this resolution
+automatically.
```
Comment (joncinque) [inline token/transfer-hook-interface/README.md:121]: Rather than copying this, how about linking to the other README?

**Review by joncinque — DISMISSED** [review]
Just some tiny nits, all good after that!

File: docs/src/token-2022/extensions.mdx (hunk 2 above)
Comment (buffalojoec) [reply docs/src/token-2022/extensions.mdx:1413]: Coooooming right up next!

**Review by joncinque — APPROVED** [review]
Great job!

**Pull request merged**